util/jsonutil/keywordsutil.go

```diff
@@ -0,0 +1,59 @@
+package jsonutil
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type KeywordsString string
+
+// extImpAppnexusKeyVal defines the contract for bidrequest.imp[i].ext.prebid.bidder.appnexus.keywords[i]
+type extImpAppnexusKeyVal struct {
+	Key    string   `json:"key,omitempty"`
+	Values []string `json:"value,omitempty"`
+}
+
+func (ks *KeywordsString) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+
+	switch b[0] {
+	case '{':
+		var results map[string][]string
+		err := json.Unmarshal(b, &results)
+		if err != nil {
+			return err
+		}
+
+		var keywords string
+		for key, values := range results {
+			for _, val := range values {
+				keywords = keywords + fmt.Sprintf("%s=%s,", key, val)
+			}
+		}
+		*ks = KeywordsString(strings.TrimSuffix(keywords, ","))
+	case '[':
+		var results []extImpAppnexusKeyVal
+		err := json.Unmarshal(b, &results)
+		if err != nil {
+			return err
+		}
```

Comment:
just a suggestion: could inline error check

```
if err := json.Unmarshal(b, &results);  err != nil {
	return err
}
```

Reply:
addressed in [f8c896a](https://github.com/prebid/prebid-server/pull/2649/commits/f8c896a4dd6fc8cd1e033a610ab939b3602e7fad)